### PR TITLE
Fixes for Moodle 4.1

### PR DESCRIPTION
**Pull request by sbourget**
Pull request includes:

- Codechecker fixes
- PHPDocs fixes
- Update for GHA to support newer Moodle versions

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/nbproject/private/
```
Comment (xow) [inline .gitignore:1]: What's this about? After a quick google maybe mplab x?

Comment (xow) [reply .gitignore:1]: Ah yep I see the accidental file

File: nbproject/project.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://www.netbeans.org/ns/project/1">
+    <type>org.netbeans.modules.php.project</type>
+    <configuration>
+        <data xmlns="http://www.netbeans.org/ns/php-project/1">
+            <name>xow/moodle-mod_quizgame</name>
+        </data>
+    </configuration>
+</project>
```
Comment (xow) [inline nbproject/project.xml:9]:
```suggestion
```

**Review by xow — APPROVED** [review]
Thanks, sorry about the delay. I wonder if we need all the netbeans stuff - but it won't do any harm

File: nbproject/project.xml (hunk 2 above)
Comment (sbourget) [reply nbproject/project.xml]: Looks like I missed those, should now be removed

**Pull request merged**